cmd/create/cluster/cmd.go

```diff
@@ -2234,13 +2238,17 @@ func run(cmd *cobra.Command, _ []string) {
 		// Verify subnets provided exist.
 		if subnetsProvided {
 			for _, subnetArg := range subnetIDs {
-				verifiedSubnet := false
-				for _, subnet := range subnets {
-					if awssdk.ToString(subnet.SubnetId) == subnetArg {
-						verifiedSubnet = true
-					}
+				// Check if subnet is in the excluded list of public subnets
+				if slices.Contains(excludedSubnetIdsDueToBeingPublic, subnetArg) {
+					r.Reporter.Errorf("The command cannot be executed because %s is public and the cluster is set as private",
```

Comment:
I would change this error message slightly, something like:
`Cluster is set as private, cannot use public subnet '%s'`

Reply:
changed it here [ed05f9e](https://github.com/openshift/rosa/pull/2832/commits/ed05f9ea971ac0d516c5f142dad1cb49cd25d7ed)